python/tk_multi_loader/model_latestpublish.py

```diff
@@ -79,107 +79,129 @@ def load_data(self, item, child_folders, show_sub_items, additional_sg_filters):
-                # it may 'pause' execution briefly for the user
-                data = app.shotgun.find(entity_type, partial_filters)
-
-
-                # now create the final query for the model - this will be
-                # a big in statement listing all the ids returned from
-                # the previous query, asking the model to only show the
-                # items matching the previous query.
-                #
-                # note that for tasks, we link via the task field
-                # rather than the std entity link field
-                #
-                if entity_type == "Task":
-                    sg_filters = [["task", "in", data]]
-                elif entity_type == "Version":
-                    sg_filters = [["version", "in", data]]
-                else:
-                    sg_filters = [["entity", "in", data]]
-
-                # lastly, when we are in this special mode, the main view
-                # is no longer functioning as a browsable hierarchy
-                # but is switching into more of a paradigm of an inverse
-                # database. Indicate the difference by not showing any folders
-                child_folders = []
+                # loop through the selected items
+                for item in items:
+                    # special mode -- in this case we don't show any of the
+                    # child folders and only the partial matches of all the leaf nodes
+
+                    # for example, this may return
+                    # entity type shot, [["sequence", "is", "xxx"]] or
+                    # entity type shot, [["status", "is", "ip"]] or
+
+                    # note! Because of nasty bug https://bugreports.qt-project.org/browse/PYSIDE-158,
+                    # we cannot pull the model directly from the item but have to pull it from
+                    # the model index instead.
+                    model_idx = item.index()
+                    model = model_idx.model()
+                    partial_filters = model.get_filters(item)
+                    entity_type = model.get_entity_type()
+
+                    # now get a list of matches from the above query from
+                    # shotgun - note that this is a synchronous call so
+                    # it may 'pause' execution briefly for the user
+                    data = self.app.shotgun.find(entity_type, partial_filters)
+
+                    # now create the final query for the model - this will be
+                    # a big in statement listing all the ids returned from
+                    # the previous query, asking the model to only show the
+                    # items matching the previous query.
+                    #
+                    # note that for tasks, we link via the task field
+                    # rather than the std entity link field
+                    #
+
+                    # New sg_filter for tags. We need to pull the tag applied
```

Comment:
suggest removing the "New " here and treat this as a note like the comment above.